addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,218 @@
+        ["WAR"] = 1,
+        ["MNK"] = 2,
+        ["WHM"] = 3,
+        ["BLM"] = 4,
+        ["RDM"] = 5,
+        ["THF"] = 6,
+    }
+    for index,value in pairs(starting_jobs) do
+        if not find_conflict(index) then 
+            return value
+        end
+    end
+end
+
+function find_job(job)
+    local self = windower.ffxi.get_player()
+    local ki = windower.ffxi.get_key_items() 
+    
+    for index,value in pairs(res.jobs) do
+        if value.ens:lower() == job then 
+            jobid = index
+        end
+    end
+    --[[
+        Small rant/request/can't figure out a better way.
+        windower.ffxi.get_player().jobs includes all jobs regardless if you have it unlocked.  I expected self.jobs["GEO"] to be nil if I didn't have it.  For now going to use a list of the KI's for 
+        Job emotes to see which jobs are unlocked. 
+    ]]    
+    local job_gesture_ids = {
+        -- Pulled from resources. 12/26/2016
+        ["WAR"] = 1738,
+        ["MNK"] = 1739,
+        ["WHM"] = 1740,
+        ["BLM"] = 1741,
+        ["RDM"] = 1742,
+        ["THF"] = 1743,
+        ["PLD"] = 1744,
+        ["DRK"] = 1745,
+        ["BST"] = 1746,
+        ["BRD"] = 1747,
+        ["RNG"] = 1748,
+        ["SAM"] = 1749,
+        ["NIN"] = 1750,
+        ["DRG"] = 1751,
+        ["SMN"] = 1752,
+        ["BLU"] = 1753,
+        ["COR"] = 1754,
+        ["PUP"] = 1755,
+        ["DNC"] = 1756,
+        ["SCH"] = 1757,
+        ["GEO"] = 2963,
+        ["RUN"] = 2964,
+    }
+    local job_gesture = job_gesture_ids[job:upper()]
+    for i,v in pairs(ki) do
+        if v == job_gesture then
+            return jobid
+        end
+    end
+end
+
+
+function find_job_change_npc()
+    local found = nil
+    local valid_zones = { 
+        -- Zones with a nomad moogle / green thumb moogle, taken from Resources
+        -- All other zones check if mog_house
+        [26] = {id=26,en="Tavnazian Safehold",ja="タブナジア地下壕",search="TavSafehld"},
+        [53] = {id=53,en="Nashmau",ja="ナシュモ",search="Nashmau"},
+        [247] = {id=247,en="Rabao",ja="ラバオ",search="Rabao"},
+        [248] = {id=248,en="Selbina",ja="セルビナ",search="Selbina"},
+        [249] = {id=249,en="Mhaura",ja="マウラ",search="Mhaura"},
+        [250] = {id=250,en="Kazham",ja="カザム",search="Kazham"},
+        [252] = {id=252,en="Norg",ja="ノーグ",search="Norg"},    
+    }
+    local info = windower.ffxi.get_info()
+    if not (valid_zones[info['zone']] or info['mog_house']) then
+        windower.add_to_chat(4,'JobChange: Not in a zone with a Change NPC')
+        return
+    end
+    for i,v in pairs(windower.ffxi.get_mob_array()) do
+        if v['name'] == 'Moogle' or v['name'] == 'Nomad Moogle' or v['name'] == 'Green Thumb Moogle' then
+            found = 1
+            target_index = i
+            target_id = v['id']
+            distance = windower.ffxi.get_mob_by_id(target_id).distance
+            if math.sqrt(distance)<6 then 
```

Comment:
As I wrote before, it's a common trick to save yourself one square root operation to just square the other side of the comparison:
```lua
        if distance < 36 then
```